pilot/pkg/model/service.go

```diff
@@ -418,6 +425,9 @@ type IstioEndpoint struct {
 
 	// Determines the discoverability of this endpoint throughout the mesh.
 	DiscoverabilityPolicy EndpointDiscoverabilityPolicy `json:"-"`
+
+	// The name of the node where the endpoint(pod) is located
+	NodeName string
```

Comment:
How does this work for non-k8s endpoints?

Reply:
node local traffic policy can not work on non k8s endpoint, as for a vm. At most one workload there

Reply:
But should be careful, is this empty for a workload?

Reply:
> But should be careful, is this empty for a workload?

yes i will be consider this